pkg/solace/config/endpoint_properties_test.go

```diff
@@ -0,0 +1,75 @@
+// pubsubplus-go-client
+//
+// Copyright 2021-2024 Solace Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"solace.dev/go/messaging/pkg/solace/config"
+)
+
+var endpointProperties = config.EndpointPropertyMap{
```

Comment:
Is there a reason for not testing all of the props we have?

Reply:
yes. The remaining props are either boolean or uint values and since Go is a strongly typed language, there is no need to have unit tests for boolean/unit values (I cannot test an invalid boolean/unit argument). The other testing is covered by integration tests that will come in another PR.

Reply:
This test isn't testing invalid values though, it is the serialization/deserialization
Even in this test `EndpointPropertyMaxMessageRedelivery` is being set as a float in your test but it should be a uint in the actual type...

Reply:
You are right, it is actually a serialization/deserialization test. I checked again and the initial comment I made was for another file. I thought it was for something else. I have updated the float64() to uint64()

Reply:
So then should we add all the other props here as well to ensure they can be serialized?

Reply:
Okay sure, make sense. I will update this with the full properties list

Reply:
Done